Functional testing

rgazelot opened the pull request:
- Add Behat

rgazelot committed:
Init behat

rgazelot committed:
Create the base : FeatureContext, SheetContext and TableContext

Add the first important steps

rgazelot commented:
Tu pourras regarder vite fait si mon environnement de test semble cohérent @Taluu, grand gourou de behat

rgazelot committed:
Add the Workbook properties into behat context [ci skip]

rgazelot committed:
Create StyleContext

rgazelot committed:
Add style to Workbook

rgazelot committed:
Change getColumnDimension by getcolumnDimensionBycolumn into Sheet

rgazelot committed:
Add SheetContext [ci skip]

rgazelot committed:
Add missing styles on Table::setColumn

rgazelot committed:
Finish the important steps

rgazelot committed:
Remove construct in Tablecontext

Taluu commented on features/bootstrap/FeatureContext.php:
```diff
@@ -0,0 +1,90 @@
+use ExcelAnt\PhpExcel\Workbook,
+    ExcelAnt\PhpExcel\Sheet;
+
+require_once 'PHPUnit/Autoload.php';
```
vire tous les require PHPUnit/*, ils servent à rien

rgazelot replied:
Bha pour l'instant ! Les tests "When" et  "Then" ne sont pas fait. Et les assetions se font a ce moment là !

Sent from my iPhone

On 8 juin 2013, at 16:07, Baptiste Clavié notifications@github.com wrote:

> In features/bootstrap/FeatureContext.php:
> 
> > @@ -0,0 +1,90 @@
> > +<?php
> > +
> > +use Behat\Behat\Context\ClosuredContextInterface,
> > -    Behat\Behat\Context\TranslatedContextInterface,
> > -    Behat\Behat\Context\BehatContext,
> > -    Behat\Behat\Exception\PendingException;
> >   +use Behat\Gherkin\Node\PyStringNode,
> > -    Behat\Gherkin\Node\TableNode;
> >   +
> >   +use ExcelAnt\PhpExcel\Workbook,
> > -    ExcelAnt\PhpExcel\Sheet;
> >   +
> >   +require_once 'PHPUnit/Autoload.php';
> >   vire tous les require PHPUnit/*, ils servent à rien
> 
> —
> Reply to this email directly or view it on GitHub.

Taluu replied:
Dans ce cas, puisque phpunit est dans ton composer, vire ces lignes, tu y auras déjà accès... Là, le problème, c'est que tes tests **ne peuvent pas fonctionner** si tu n'as pas installé phpunit via pear, ou si tu l'as pas en accès global. Ce que la plupart des configs ont.

De toutes manières, cette inclusion des asertions phpUnit ont été dépréciées dans la doc (je le sais, c'est moi qui ait poussé ce changement, d'utiliser composer à la place...)

rgazelot committed:
Change the responsibility of the Writer. There are now two methods : convert and write

rgazelot committed:
First Feature

Write the behat.xls for the tests

rgazelot committed:
Change the position of the test about the sheet number

Add Asserts in all Context file

rgazelot committed:
Remove PHPExcel in WriterInterface

rgazelot committed:
Add backslashes before all global classes

rgazelot committed:
Change the organization of PhpExcel Workbook and Sheet

rgazelot committed:
Add PhpExcel into Adapter folder

rgazelot committed:
Add Adapter before PhpExcel path [ci skip]

rgazelot committed:
Add workbook properties scenario

rgazelot committed:
Create a simple export to be easily tested

Add default title on new sheet

Add style when we add a new cell

Add Then step to check the style of a specific cell in the export

Add new workbook scenario

Add type property for the Fill style

rgazelot committed:
Move the behat tests into PHPExcel folder

rgazelot committed:
Add Behat in travis

rgazelot committed:
Add progress format to behat test-suite

rgazelot committed:
Write the Use with the right way

rgazelot committed:
Behat : Two tables in one sheet

Behat : All label (top,left,full)
